src/models/event.js

```diff
@@ -721,7 +749,7 @@ utils.extend(module.exports.MatrixEvent.prototype, {
      * @return {boolean} True if this event has been redacted
      */
     isRedacted: function() {
-        return Boolean(this.getUnsigned().redacted_because);
+        return this._locallyRedacted || Boolean(this.getUnsigned().redacted_because);
```

Comment:
`markLocallyRedacted` seems to also set `redacted_because`, so perhaps we don't need to check `this._locallyRedacted` here?